pkg/packages/util.go

```diff
@@ -15,22 +16,30 @@ func ExtractErrors(pkg *packages.Package) []packages.Error {
 	seenErrors := map[string]bool{}
 	var uniqErrors []packages.Error
 	for _, err := range errors {
-		if seenErrors[err.Msg] {
+		msg := stackCrusher(err.Error())
+		if seenErrors[msg] {
 			continue
 		}
-		seenErrors[err.Msg] = true
+
+		if msg != err.Error() {
+			continue
+		}
+
+		seenErrors[msg] = true
+
 		uniqErrors = append(uniqErrors, err)
 	}
 
 	if len(pkg.GoFiles) != 0 {
-		// errors were extracted from deps and have at leat one file in package
+		// errors were extracted from deps and have at least one file in package
 		for i := range uniqErrors {
-			_, parseErr := ParseErrorPosition(uniqErrors[i].Pos)
-			if parseErr != nil {
-				// change pos to local file to properly process it by processors (properly read line etc)
-				uniqErrors[i].Msg = fmt.Sprintf("%s: %s", uniqErrors[i].Pos, uniqErrors[i].Msg)
-				uniqErrors[i].Pos = fmt.Sprintf("%s:1", pkg.GoFiles[0])
+			if _, parseErr := ParseErrorPosition(uniqErrors[i].Pos); parseErr == nil {
```

Comment:
Tiniest of nits but is there a reason flipping this logic? Maybe it's not a convention but I tend to keep happy path to the left. 😸

Reply:
In this case, `parseErr` must be seen as a simple output more than an error.
An also, it decreases cognitive complexity: less indentation and skips early.

Reply:
Thanks for explaining, that makes sense! Your arguments are exactly why I like [happy path to the left](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88), to get less complexity and use early returns.  But this makes more sense in this loop where we only want to act on the errors.